src/NewProject/add_constraints_form.cpp

```diff
@@ -22,6 +26,17 @@ addConstraintsForm::addConstraintsForm(QWidget *parent)
 
   ui->m_ckkBoxCopy->setText(tr("Copy sources into project."));
   ui->m_ckkBoxCopy->setCheckState(Qt::CheckState::Checked);
+
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (compiler->PinAssignOpts() == Compiler::PinAssignOpt::Random)
+    ui->select_random->setChecked(true);
+  else
+    ui->select_defineOrder->setChecked(true);
+
+  connect(ui->select_defineOrder, &QRadioButton::clicked, this,
```

Comment:
critical issue is here. User can press Cancel but the selection will take affect. User selection need to buffer somewhere and apply to compiler only after user press Ok.

Reply:
Okay